src/careamics/config/vae_algorithm_model.py

```diff
@@ -0,0 +1,118 @@
+    training algorithm: which algorithm, loss function, model architecture, optimizer,
+    and learning rate scheduler to use.
+
+    Currently, we only support N2V, CARE, N2N and custom models. The `n2v` algorithm is
+    only compatible with `n2v` loss and `UNet` architecture. The `custom` algorithm
+    allows you to register your own architecture and select it using its name as
+    `name` in the custom pydantic model.
+
+    Attributes
+    ----------
+    algorithm : Literal["n2v", "custom"]
+        Algorithm to use.
+    loss : Literal["n2v", "mae", "mse"]
+        Loss function to use.
+    model : Union[UNetModel, LVAEModel, CustomModel]
+        Model architecture to use.
+    optimizer : OptimizerModel, optional
+        Optimizer to use.
+    lr_scheduler : LrSchedulerModel, optional
+        Learning rate scheduler to use.
+
+    Raises
+    ------
+    ValueError
+        Algorithm parameter type validation errors.
+    ValueError
+        If the algorithm, loss and model are not compatible.
+
+    Examples
+    --------
+    # TODO add once finalized
+    """
+
+    # Pydantic class configuration
+    model_config = ConfigDict(
+        protected_namespaces=(),  # allows to use model_* as a field name
+        validate_assignment=True,
+        extra="allow",
+    )
+
+    # Mandatory fields
+    # defined in SupportedAlgorithm
+    algorithm_type: Literal["vae"]
+    algorithm: Literal["musplit", "denoisplit", "custom"]
+    loss: Literal["musplit_loss", "denoisplit_loss"]
+    model: Union[LVAEModel, CustomModel] = Field(discriminator="architecture")
+
+    noise_model: GaussianMixtureNmModel = Field(discriminator="model_type")
+    noise_model_likelihood_model: Optional[NMLikelihoodModel] = None
+    gaussian_likelihood_model: Optional[GaussianLikelihoodModel] = None
+
+    # Optional fields
+    optimizer: OptimizerModel = OptimizerModel()
+    """Optimizer to use, defined in SupportedOptimizer."""
+
+    lr_scheduler: LrSchedulerModel = LrSchedulerModel()
+
+    @model_validator(mode="after")
+    def algorithm_cross_validation(self: Self) -> Self:
+        """Validate the algorithm model based on `algorithm`.
+
+        Returns
+        -------
+        Self
+            The validated model.
+        """
+        # musplit
+        if self.algorithm == "musplit":
+            if self.loss != "musplit_loss":
+                raise ValueError(
+                    f"Algorithm {self.algorithm} only supports loss `musplit`."
+                )
+            if self.model.predict_logvar != "pixelwise":
+                raise ValueError(
+                    "Algorithm `musplit` only supports `predict_logvar` as `pixelwise`."
+                )
+        # TODO add more checks
+
+        if self.algorithm == "denoisplit":
+            if self.loss == "denoisplit_loss":
+                raise ValueError(
+                    f"Algorithm {self.algorithm} only supports loss `denoisplit_loss`."
+                )
```

Comment:
Also we should probably be comparing with the `Enum` class values here rather than strings. It might have saved some trouble. `SupportedLoss.MUSPLIT=="musplit"` will return `True`.